tests/fixtures/ctx.rs

```diff
@@ -297,6 +297,11 @@ impl E2ECtx {
         })
     }
 
+    pub async fn stop_all(&mut self) -> Result<()> {
+        futures::future::join_all(self.nodes.iter_mut().map(|node| node.stop())).await;
+        Ok(())
+    }
```

Comment:
don't need a `Result` here I think

```suggestion
    pub async fn stop_all(&mut self) {
        futures::future::join_all(self.nodes.iter_mut().map(|node| node.stop())).await;
    }
```